design-toolkit/components/src/components/tree/index.tsx

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { TreeStyles, TreeStylesDefaults } from '@/components/tree/styles';
+import type { TreeNode } from '@/hooks/types';
+import { isSlottedContextValue } from '@/lib/utils';
+import { DragVert } from '@accelint/icons';
+import type { Key } from '@react-types/shared';
+import {
+  type PropsWithChildren,
+  createContext,
+  memo,
+  useContext,
+  useMemo,
+} from 'react';
+import {
+  Text as AriaText,
+  Tree as AriaTree,
+  TreeItem as AriaTreeItem,
+  TreeItemContent as AriaTreeItemContent,
+  type ContextValue,
+  DropIndicator,
+  type DropTarget,
+  composeRenderProps,
+  useDragAndDrop,
+} from 'react-aria-components';
+import { Button } from '../button';
+import { Icon } from '../icon';
+import { Lines } from '../lines';
+import { ExpandToggle } from './expand-toggle';
+import { SelectionToggle } from './selection-toggle';
+import type {
+  ItemContentProps,
+  ItemContentRenderProps,
+  ItemTextProps,
+  TreeItemProps,
+  TreeProps,
+} from './types';
+import { VisibilityToggle } from './visibility-toggle';
+
+const {
+  tree,
+  item,
+  content,
+  display,
+  icon,
+  label,
+  actions,
+  spacing,
+  description,
+  drag,
+} = TreeStyles();
+
+export const TreeContext =
+  createContext<
+    ContextValue<TreeProps<unknown> & { isStatic?: boolean }, HTMLDivElement>
+  >(null);
+
+const defaultRenderDropIndicator = (target: DropTarget) => (
+  <DropIndicator target={target} className='border border-highlight-hover' />
+);
+
+const TreeLines = memo(function TreeLines({
```

Comment:
Is there a specific need to memoize this? Can we lean on the compiler to handle this?